core/src/data_sources/data_source.rs

```diff
@@ -588,7 +588,13 @@ impl DataSource {
         text: Section,
         preserve_system_tags: bool,
     ) -> Result<Document> {
-        let full_text = text.full_text();
+        // converting NULL byte ('\0') to an empty string ("")
```

Comment:
This won't work in general as you'll need to do a recursive replacement.

Ooc when are the \0 bytes causing problem? I recall us already removing them when chunking?

Reply:
Your change may also interfere with the hashing logic below.

Reply:
You mention Postgres, what is the impacted table?

Reply:
If my understanding is correct we are having issues when upserting in the `data_sources_documents` table.

Reply:
I guess you are thinking of this PR ? https://github.com/dust-tt/dust/pull/5598/files

Reply:
data sources document dont store the content? Do you have the error handy?

Reply:
```
Upsert error: Failed to upsert document (error: db error: ERROR: invalid byte sequence for encoding "UTF8": 0x00

Caused by:
    ERROR: invalid byte sequence for encoding "UTF8": 0x00)
```

Reply:
I guess it's occuring in one of the metadata field?